storage/store/headers.go

```diff
@@ -21,34 +21,83 @@ type Headers struct {
 	heightCache *Cache[uint64, flow.Identifier]
 	viewCache   *Cache[uint64, flow.Identifier]
 	sigs        *proposalSignatures
+	chainID     flow.ChainID
 }
 
 var _ storage.Headers = (*Headers)(nil)
 
 // NewHeaders creates a Headers instance, which stores block headers.
-// It supports storing, caching and retrieving by block ID and the additionally indexed by header height.
-func NewHeaders(collector module.CacheMetrics, db storage.DB) *Headers {
-	storeWithLock := func(lctx lockctx.Proof, rw storage.ReaderBatchWriter, blockID flow.Identifier, header *flow.Header) error {
-		return operation.InsertHeader(lctx, rw, blockID, header)
-	}
-
-	retrieve := func(r storage.Reader, blockID flow.Identifier) (*flow.Header, error) {
-		var header flow.Header
-		err := operation.RetrieveHeader(r, blockID, &header)
-		return &header, err
+// It supports storing, caching and retrieving by block ID, and additionally indexes by header height and view.
+func NewHeaders(collector module.CacheMetrics, db storage.DB, chainID flow.ChainID) *Headers {
```

Comment:
Please document the requirements on the `chainID` input - I would reference the constant `ClusterIDPrefix` for further reading.